### DISCUSSION TITLE
Let's get this a solid community

### DISCUSSION THREAD

**Discussion by NullPlane** [SpartanJ/ecode#324] [General]
I love ecode and Ig there's more than just me.
Let's share this project more often and get it a 
bigger community 👍 we're on the runway to
1k

[![image](https://img.shields.io/github/stars/SpartanJ/ecode?style=for-the-badge&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CPHN2ZyB3aWR0aD0iMjM0IiBoZWlnaHQ9IjMyNCIgdmVyc2lvbj0iMS4xIiB2aWV3Qm94PSIwIDAgNS43OTk1IDguMDMzNyIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKC0zLjQ1MDIgLTIuMzIyMSkiPgogIDxnIHRyYW5zZm9ybT0ibWF0cml4KC44MDU4OCAwIDAgLjgwNTg4IDEuMzY0MSAxLjI1MzgpIiBmaWxsPSJub25lIiBzdHJva2U9IiNmZmYiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLXdpZHRoPSIuOHB4Ij4KICAgPHBhdGggdHJhbnNmb3JtPSJtYXRyaXgoMSAwIDAgLTEgLjE5ODA4IDExLjk4NCkiIGQ9Im02LjA5NDMgNS45ODk5YzAuMTY0NTcgMC4xMTczIDAuMDA5ODc4NSAwLjM1OTk3LTAuMTExNDUgMC40MzYxMi0wLjQwODQ4IDAuMjU2MzgtMC44OTAyMi0wLjExNjIyLTEuMDM5LTAuNDk2NS0wLjMyOTgzLTAuODQyOTkgMC4zNzkzMi0xLjY3OTQgMS4xNjQ2LTEuODk2NyAxLjQyMTEtMC4zOTMzOCAyLjcyOTUgMC43NzkwNCAzLjAwOTMgMi4xMTU4IDAuMzY4MjIgMS43NTkzLTAuNzk1MzcgMy40NTc2LTIuMzk4OSA0LjEwOTYiLz4KICAgPHBhdGggdHJhbnNmb3JtPSJtYXRyaXgoMSAwIDAgLTEgLjAzMDI4NCAxMi4yNTIpIiBkPSJtNi4zMTYzIDYuMzYwOGMtMC4xNDY4OS0wLjE4NTUyIDAuMTA2MTEtMC40MDgwNiAwLjI2NzM3LTAuNDU0MTkgMC41NDI4OS0wLjE1NTMyIDAuOTYzMzUgMC40MjA2MSAxLjAwNjggMC44OTY3NCAwLjA5NjQyNCAxLjA1NTUtMC45NzM0OSAxLjc2NTItMS45Mjc1IDEuNzUzOS0xLjcyNjMtMC4wMjAzNjgtMi44MTYxLTEuNzY1LTIuNjk1NC0zLjM1OTUgMC4xNTg4MS0yLjA5ODYgMi4wMjA1LTMuNjI5NyA0LjAzNjMtMy44NDA2Ii8%2BCiAgPC9nPgogPC9nPgo8L3N2Zz4K&logoColor=2aa0da&logoSize=auto&label=stars4ecode%F0%9F%92%9A&color=2aa0da&cacheSeconds=15000)](https://github.com/SpartanJ/ecode/stargazers "Give him a little honorful 'thanks'")

**Comment by SpartanJ** [comment]
I should do my part but I honestly don't participate in social media and I'm not good at marketing hehe. Also, at least until now I didn't want to promote much because more exposure translate into more user demands and issues open, which will distract me from my main pending objetives (basically it's just the debugger support that is preventing me to switch completely to ecode, I need to be able to debug ecode from ecode 😆).
Things I know users will insist to have if ecode gets more popular:
* regex support in all searches (planned)
* a proper settings panel (planned but I have no hurry, I barely change any setting once it's configured, so I don't care)
* better plugin settings / configuration (same as above)
* better documentation (required)
* Remote development support (it would be cool, but I honestly I don't care about it, this is a blocker for many people, required changes to support this are quite large)
* Modal editing / vim keybindings (I've never used them so I really don't care about it)
* Debugger (planned, absolutely fundamental IMO, but I also know that many people don't require or need it)
* real auto-indentation (it's a mystery to me why do people care about this one)
* improved syntax-highlighting and probably tree-sitter support (these are just incremental improvements so they will gradually improve, user collaboration for this is really easy too)
... and much more, ofc

So there are several things pending that some users will find blockers, I cannot make everyone happy though, so I shouldn't care that much. Currently ecode is almost perfect for my needs, so I guess there could be plenty of people who would find ecode nice for their usage.

In conclusion, please feel free to share and mention ecode whenever you find it relevant, I'll be happy to see the project grow. Also, if anyone has ideas that can help to increase the exposure of the project without being spammy, they are welcome.

**Reply by NullPlane** [reply]
So it's to be a self-sustainer soon 👍 Yes there is some features to still be missing 
but I think it's gonna be great... From what ecode is now it looks like there is HUGE
potential for more journeymanship because by how it's made there is so much 
room for crazy stuff no one did before. I also rarely ever see software using a 2DGE 
as their framework...

I think that documentation and plugin extensibility may become a crazy tool and 
I'd love to do something if I ever get the chance to learn cpp in my carreer... It is a
problem by now but still the entire eepp project seems very exciting. Idk how easy
it is to maintain the multi-platforming but I've never to date seen someone ship a
haiku-native version of something ex works XD. And I think with some popularity
it is a great resource for creating some interfaces for many use cases. I could 
literally see a perfect use case for some UI in data analysis, esp. live~.

Modal editing may be a great feature. Maybe a little journeymanship and something
exclusive that you find useful and want to experiment on. Modes by activation or 
separate windows/popups may also be a thing... But tbh. for ***partial*** VIM keybinds, 
OK nice, but I've never understood that ppl want that in such editors. If you want a VIM 
experience you should be using TUI Editors anyway so what you're doing here, it sounds 
strange to me...

I see a lot for exclusive ideas. Somehow this editor felt new and exciting to me in some 
way. Idk what you did differently but it was like a stripped down version of the tropically
mundane ones used everywhere but actually made usable and compressed to the needs
of someone who does not want to waste 5000hrs of time on their software or system... 



For the advertisement thingy btw I am actually a bit happy that ecode is more niche... 
I got the feeling that anything using social media for advertisement screams "inherently
bad product" to me nowadays. And I think a lot of ppl that do not belong to late Gen Z 
or Alpha will or do feel like that... There's too great of a mess over there on modern SoMe.
